### ISSUE TITLE
md-button not stretching in flex column layout (0.10.0)

### ISSUE THREAD

**Issue by chrismbeckett** [angular/material#3589]
Got some buttons in a form that I want to be full width across the container. I have put my button in a flex column layout and set each button for flex. They should stretch, they don't. Worked in 0.9.8, broken in 0.10.0. Works if layout is row instead of column.

```html
<div layout="column">
  <md-button class="md-raised" flex>Button 1</button>
</div>
```

The problem seems to be the ``display: inline-table``. Fixed it with this:

```sass
[layout='column'] {
  .md-button {
    display: block;
  }
}
```

**Comment by Splaktar** [comment]
It seems to be working properly. Here's a CodePen that demonstrates it: http://codepen.io/Splaktar/pen/WvzrWY

I gave a few examples that might help you figure out what is happening in your app. Note that if your `layout-align` is set to certain values, the `flex` on a `layout="column" might only expand vertically.

**Comment by chrismbeckett** [comment]
Hey, thanks for the Plunker. After playing around with it, the problem seems to be a conflict with Angular UI Router. I omitted it from my sample above because I had never considered that it would be the source of the problem. If I omit the **ui-sref** then the buttons do flex stretch properly. As soon as I add either an **href** or **ui-sref** property, they no longer stretch.

```js
<md-button type="button" class="md-raised" ui-sref="account.login">
  {{vm.labels.login}}
</md-button>
```

**Comment by Splaktar** [comment]
Maybe you could wrap the button with an anchor containing the `ui-sref`. Something like that was mentioned in #2131. Wrapping with an anchor is probably better than a `div` (mentioned in that issue) for accessibility.

**Comment by chrismbeckett** [comment]
Unlike #2131 where a list item that already had an aria role set was being used for a link, this is buttons vs. links - no accessibility conflict here. The Angular Material site button demos page shows buttons being used as links which is very common, and on the demo page, the buttons with ``ng-href`` attributes are rendered as ``<a>`` links.

https://material.angularjs.org/latest/#/demo/material.components.button

I just don't think that flexbox likes ``<a>`` elements with ``display: inline-table``.

**Comment by marcysutton** [comment]
Just a note that you can't wrap interactive elements within each other: a button can't go inside of a link.

**Comment by dietergeerts** [comment]
the md-button also doesn't remove it's margin, which makes it un aligned with the rest in a form.

**Comment by topherfangio** [comment]
Can you verify if this is still an issue? Based off of the Codepen in #3227 it looks like the `<a>` and `<button>` tags now both use `display: inline-block` which should resolve this issue.

**Comment by chrismbeckett** [comment]
Yes, I can confirm, this is fixed in 0.11.0.
